### ISSUE TITLE
Setting weight by input message. 

### ISSUE THREAD

**Issue by rharrex** [bartbutenaers/node-red-contrib-msg-router#5]
Hi Bart,
 I'm having trouble setting the weight using the msg.weight field. There appears to be two different spellings in your code "weight" and weigth. This maybe by design. 
Regards,
Rod

**Comment by rharrex** [comment]
Test setup and image file demonstrating issue. Tested on both Pi4 and windows 10, version of NR 20.8 and 1.00beta2 respectively.
[{"id":"8dcd40af.6c5a7","type":"change","z":"4d0ff0dd.2e38c8","name":"","rules":[{"t":"set","p":"output","pt":"msg","to":"2","tot":"num"},{"t":"set","p":"active","pt":"msg","to":"true","tot":"bool"}],"action":"","property":"","from":"","to":"","reg":false,"x":520,"y":600,"wires":[["c0a32541.4272c","ab9b097a.08b73"]]}]

![image](https://user-images.githubusercontent.com/56061692/66277643-1fce0780-e8ed-11e9-84b1-c1e3d36f1aea.png)

**Comment by bartbutenaers** [comment]
Hey Rod (@rharrex ),

Thanks for reporting this issue!  Every time I have to write English, I'm always doubting if it should be '_weight_' or '_weigth_' ;-)  Normally the consequences of this typo are not that big, until now ...

Have fixed it on Github, and updated the info panel and readme file.  

Your test flow only contained only a single Change-node, so I could not use it for testing.  Would be nice if you could test it, before I publish it on NPM.  You can install it directly from Github like this:
```
npm install bartbutenaers/node-red-contrib-msg-router
```

I created quickly following test flow, which toggles a weight of ```0``` or ```100``` percent:

![image](https://user-images.githubusercontent.com/14224149/66342932-d4e2dd00-e94a-11e9-9f28-7f51c9e74115.png)

```
[{"id":"e8954926.0eeea8","type":"msg-router","z":"dc82a538.a81ba8","routerType":"weightedroundrobin","topicDependent":false,"counterReset":false,"msgKeyField":"payload","undefinedHash":false,"outputsInfo":[{"active":true,"clone":false,"delay":"0","weight":"100"},{"active":true,"clone":false,"delay":"0","weight":"0"}],"name":"","delaying":"unrelated","msgControl":true,"outputs":2,"x":830,"y":600,"wires":[["b78a3534.3499e8"],["93784955.aa9548"]]},{"id":"48ba6a5a.2190c4","type":"inject","z":"dc82a538.a81ba8","name":"Message to be routed","topic":"","payload":"","payloadType":"date","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":540,"y":520,"wires":[["e8954926.0eeea8"]]},{"id":"389810ed.27dee","type":"inject","z":"dc82a538.a81ba8","name":"All to output 1","topic":"","payload":"","payloadType":"date","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":350,"y":600,"wires":[["f3f07063.736bb","6068b2b9.f5debc"]]},{"id":"f3f07063.736bb","type":"change","z":"dc82a538.a81ba8","name":"Output 1 to 100","rules":[{"t":"set","p":"output","pt":"msg","to":"1","tot":"num"},{"t":"set","p":"weight","pt":"msg","to":"100","tot":"num"}],"action":"","property":"","from":"","to":"","reg":false,"x":560,"y":600,"wires":[["e8954926.0eeea8"]]},{"id":"6068b2b9.f5debc","type":"change","z":"dc82a538.a81ba8","name":"Output 2 to 0","rules":[{"t":"set","p":"output","pt":"msg","to":"2","tot":"num"},{"t":"set","p":"weight","pt":"msg","to":"0","tot":"num"}],"action":"","property":"","from":"","to":"","reg":false,"x":550,"y":640,"wires":[["e8954926.0eeea8"]]},{"id":"47b95849.fadb48","type":"inject","z":"dc82a538.a81ba8","name":"All to output 2","topic":"","payload":"","payloadType":"date","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":350,"y":680,"wires":[["64026a55.3a4474","d2adb1e2.5a76c"]]},{"id":"64026a55.3a4474","type":"change","z":"dc82a538.a81ba8","name":"Output 1 to 0","rules":[{"t":"set","p":"output","pt":"msg","to":"1","tot":"num"},{"t":"set","p":"weight","pt":"msg","to":"0","tot":"num"}],"action":"","property":"","from":"","to":"","reg":false,"x":550,"y":680,"wires":[["e8954926.0eeea8"]]},{"id":"d2adb1e2.5a76c","type":"change","z":"dc82a538.a81ba8","name":"Output 2 to 100","rules":[{"t":"set","p":"output","pt":"msg","to":"2","tot":"num"},{"t":"set","p":"weight","pt":"msg","to":"100","tot":"num"}],"action":"","property":"","from":"","to":"","reg":false,"x":560,"y":720,"wires":[["e8954926.0eeea8"]]},{"id":"b108d511.68db48","type":"comment","z":"dc82a538.a81ba8","name":"Default all to output 1","info":"","x":820,"y":560,"wires":[]},{"id":"b78a3534.3499e8","type":"debug","z":"dc82a538.a81ba8","name":"Output 1","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","x":1060,"y":580,"wires":[]},{"id":"93784955.aa9548","type":"debug","z":"dc82a538.a81ba8","name":"Output 2","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","x":1060,"y":620,"wires":[]}]
```

It is a pitty that you are the first one reporting this, because this means nobody has ever used it before ...

Bart

**Comment by rharrex** [comment]
Hi Bart, 
Now working as expected. Thanks very much. Appreciate your work. 
I'm using your msg-router  in an application to duty cycle multiple water pumps.  I've include some timer nodes to make round-robin decisions also based upon total run time of each pump ( to even wear on pumps) and the current running time (to ensure pumps don't overheat etc.).  Maybe you include smoothing or averaging algorithms in your msg-router. Just a thought.
Thanks again.
Rod

**Comment by bartbutenaers** [comment]
Evening Rod,
Thanks for testing!  Have published version **0.0.6** on NPM.

About the smoothing or averaging algorithms.  I'm rather busy at the moment, but if you could give me some brief explanation I could in the near future perhaps have a look at it.  It if you could find code examples (preferrable in Javascript), that would a great help to reduce development time.  

I will close this issue and you can make a new one for the algorithms, to ensure things don't get mixed...
Bart
